scala/friesian/src/test/java/RecallInitializerTest.java

```diff
@@ -0,0 +1,49 @@
+import com.intel.analytics.bigdl.friesian.nearline.recall.RecallInitializer;
+import com.intel.analytics.bigdl.friesian.nearline.utils.NearlineUtils;
+import com.intel.analytics.bigdl.friesian.serving.recall.IndexService;
+import org.apache.spark.ml.linalg.DenseVector;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class RecallInitializerTest {
+
+    @Test
+    void testInitialization() throws IOException, InterruptedException {
+        String configPath = "/home/xingyuan/projects/serving/BigDL/scala/friesian/src/test/resources/nearlineConfig/config_recall.yaml";
+        RecallInitializer.main(new String[]{"-c", configPath});
+        // you can get initialDataPath file from friesian-serving.tar.gz
+
+        SparkSession sparkSession = SparkSession.builder().getOrCreate();
+        Dataset<Row> dataset = sparkSession.read().parquet(NearlineUtils.helper().getInitialDataPath());
+
+        IndexService indexService = new IndexService(NearlineUtils.helper().getIndexDim());
+        indexService.load(NearlineUtils.helper().getIndexPath());
+
+        assertEquals(dataset.count(), indexService.getNTotal());
+        Map<Integer, double[]> vectorMap = new HashMap<>();
```

Comment:
Check NTotal is enough.

Reply:
In the following code, there is assertArrayEquals for inserted vectors.